src/libra_py/dynamics/qtag/qtag_init.py

```diff
@@ -92,13 +91,14 @@ def grid(dyn_params):
 
     params = dict(dyn_params)
 
-    critical_params = [ "nstates" ]
+    critical_params = [ "states" ]
```

Comment:
these are not parameters, these are the dynamical variables - they must be provided by the user (and that's why you feel to make it a critical  parameter)